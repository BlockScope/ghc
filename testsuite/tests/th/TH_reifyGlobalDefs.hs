-- test reification of global definitions
{-# LANGUAGE TemplateHaskell #-}
import Language.Haskell.TH.Syntax
import System.IO


g :: Int
g = 1

main :: IO ()
main =
    $(do
      let printTypeOf n = do
            addModFinalizer $ do
              VarI _ t _ <- reify (mkName n)
              runIO $ hPutStrLn stderr (n ++ " :: " ++ show t)
      printTypeOf "g"
      ds <- [d| f = True |]
      addTopDecls ds
      printTypeOf "f"
      [| return () |]
     )
