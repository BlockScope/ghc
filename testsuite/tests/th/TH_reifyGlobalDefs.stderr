f :: ConT GHC.Types.Bool
g :: ConT GHC.Types.Int
